projects/ad2s1210_iio/app/app_config.c

```diff
@@ -0,0 +1,166 @@
+/***************************************************************************//**
+ *   @file    app_config.c
+ *   @brief   Application configurations module (platform-agnostic)
+ *   @details This module performs the system configurations
+********************************************************************************
+ * Copyright (c) 2023 Analog Devices, Inc.
+ * Copyright (c) 2023 BayLibre, SAS.
+ * All rights reserved.
+ *
+ * This software is proprietary to Analog Devices, Inc. and its licensors.
+ * By using this software you agree to the terms of the associated
+ * Analog Devices Software License Agreement.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdbool.h>
+
+#include "app_config.h"
+#include "common.h"
+#include "no_os_error.h"
+#include "no_os_uart.h"
+#include "no_os_gpio.h"
+#include "no_os_irq.h"
+#include "no_os_pwm.h"
+#include "no_os_delay.h"
+
+/******************************************************************************/
+/************************ Macros/Constants ************************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/******************** Variables and User Defined Data Types *******************/
+/******************************************************************************/
+
+/* Trigger GPIO init parameters */
+struct no_os_gpio_init_param trigger_gpio_param = {
+	.port = TRIGGER_GPIO_PORT,
+	.number = TRIGGER_GPIO_PIN,
+	.pull = NO_OS_PULL_NONE,
+	.platform_ops = &trigger_gpio_ops,
+	.extra = &trigger_gpio_extra_init_params
+};
+
+/* Trigger GPIO IRQ parameters */
+struct no_os_irq_init_param trigger_gpio_irq_params = {
+	.irq_ctrl_id = 0,
+	.platform_ops = &trigger_gpio_irq_ops,
+	.extra = &trigger_gpio_irq_extra_params
+};
+
+/* PWM init parameters */
+static struct no_os_pwm_init_param pwm_init_params = {
+	.id = 0,
+	.period_ns = CONV_TRIGGER_PERIOD_NSEC,
+	.duty_cycle_ns = CONV_TRIGGER_DUTY_CYCLE_NSEC,
+	.extra = &pwm_extra_init_params,
+	.platform_ops = &pwm_ops
+};
+/* UART init parameters */
+static struct no_os_uart_init_param uart_init_params = {
+	.device_id = 0,
+	.baud_rate = IIO_UART_BAUD_RATE,
+	.size = NO_OS_UART_CS_8,
+	.parity = NO_OS_UART_PAR_NO,
+	.stop = NO_OS_UART_STOP_1_BIT,
+	.platform_ops = &uart_ops,
+	.extra = &uart_extra_init_params
+};
+
+/* UART descriptor */
+struct no_os_uart_desc *uart_desc;
+struct no_os_gpio_desc *trigger_gpio_desc;
```

Comment:
Are we using this pin as a GPIO anywhere?

Reply:
yes. in gpio_trigger_Init.  i think the pwm output is used on the gpio pin (input) as an irq,
its all muxed internally in the soc. this triggers the sampling at pwm frequency.  is my understanding correct?

Reply:
The understanding is correct, but any reason why we are specifically initializing the pin as a GPIO again?
PWM pin would be responsible for generating the signal at the frequency of set sampling rate, which would be initialized as a source of interrupt via the IRQ controller. Then do we really need the GPIO initialization for the same pin?

Please correct me if my understanding is wrong

Reply:
you are right. we dont need this gpio. Its the _gpio_ in gpio_irq that confused me a bit. i will remove this.